examples/pytorch/text-classification/run_glue.py

```diff
@@ -215,6 +215,17 @@ def main():
     else:
         model_args, data_args, training_args = parser.parse_args_into_dataclasses()
 
+    # Sending telemetry. Tracking the example usage helps us better allocate resources to maintain them. The
+    # information sent is the one passed as arguments along with your Python/PyTorch versions.
+    model_name = None if os.path.isdir(model_args.model_name_or_path) else model_args.model_name_or_path
+    if data_args.task_name is not None:
+        dataset_name = f"glue-{data_args.task_name}"
+    elif data_args.dataset_name is not None:
+        dataset_name = data_args.dataset_name
+    else:
+        dataset_name = None
+    send_example_telemetry("run_glue", model_name=model_name, dataset_name=dataset_name)
```

Comment:
I'd rather have as few lines of code as possible for the telemetry, so as to not trouble understanding. Here I understand all of that block is just for the `send_example_telemetry` thanks to the diff, but if I was reading the example in order to understand it I think I would think that `model_name` and `dataset_name` are re-used afterwards.

How about passing the `model_args`/`data_args` directly to the method? Or if you think there is too much information in there, how about defining a local method so that we understand the scope of these variables?